presto-hive/src/main/java/com/facebook/presto/hive/util/SerDeUtils.java

```diff
@@ -61,183 +66,148 @@
 
     private SerDeUtils() {}
 
-    public static byte[] getJsonBytes(DateTimeZone sessionTimeZone, Object object, ObjectInspector objectInspector)
+    public static Slice getBlockSlice(DateTimeZone sessionTimeZone, Object object, ObjectInspector objectInspector)
     {
-        ByteArrayOutputStream out = new ByteArrayOutputStream();
-        try (JsonGenerator generator = new JsonFactory().createGenerator(out)) {
-            serializeObject(sessionTimeZone, generator, object, objectInspector, null);
-        }
-        catch (IOException e) {
-            throw Throwables.propagate(e);
+        ArrayBuilder builder = ArrayBuilder.valueHolder();
+        serializeObject(sessionTimeZone, builder, object, objectInspector);
+        return builder.toSlice();
+    }
+
+    @VisibleForTesting
+    static Slice getPrimitiveSlice(DateTimeZone sessionTimeZone, Object object, ObjectInspector objectInspector)
```

Comment:
`getPrimitiveSlice` is moved to `TestSerDeUtils` but it still have to access private static method `serializeObject`